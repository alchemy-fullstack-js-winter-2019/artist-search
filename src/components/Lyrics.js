import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { getLyrics } from '../services/musicBrainzApi';
import { Link } from 'react-router-dom';

export default class Lyrics extends PureComponent {
  constructor(props) {
    super(props);
  }
  state = {
    lyrics: ''
  }

  fetchLyrics = () => {
    const artistAlter = this.props.match.params.artist.replace(/ /g, '-');
    const workAlter = this.props.match.params.work.replace(/ /g, '-');
    console.log(workAlter);
    getLyrics(artistAlter, workAlter)
      .then(res => {
        if(res.lyrics) {
          this.setState({
            lyrics: res.lyrics.split('Paroles de la chanson ')
          });
        }
        else {
          this.setState({
            lyrics: 'Could not fetch lyrics for this song. Please try again.'
          });
        }
      });
  }

  componentDidMount() {
    console.log('hello?');
    this.fetchLyrics();
  }

  render() {
    return (
      <>
        <p>{this.state.lyrics}</p>
        <Link to='/'>Return to search</Link>
      </>
    );
  }
}

Lyrics.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      artist: PropTypes.string.isRequired,
      work: PropTypes.string.isRequired
    })
  })
};
