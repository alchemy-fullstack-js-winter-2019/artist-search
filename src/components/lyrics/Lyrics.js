import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { getSongLyrics } from '../../services/artistsAndWorks';

export default class Lyrics extends PureComponent {
  state = {
    lyrics: []
  }
  static propTypes = {
    match: PropTypes.object.isRequired
  }
  componentDidMount() {
    getSongLyrics(this.props.match.params.artist, this.props.match.params.song)
      .then(response => {
        console.log(response);
        this.setState({ lyrics: response.results });
      });
  }
  render() {
    const { lyrics } = this.state;
    return (
      <Fragment>
        <h2>Song Title</h2>
        <p>{lyrics}</p>
      </Fragment>
    );
  }
}

