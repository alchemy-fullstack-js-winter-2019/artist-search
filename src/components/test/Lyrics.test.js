import React from 'react';
import Lyrics from '../Lyrics';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';


jest.mock('../../services/musicBrainzApi.js');

describe('Lyrics', () => {
  it('matches a snapshot', () => {
    const match = { params: { 
      artist: 'coldplay',
      work: 'fix-you'
    } };
    const tree = renderer.create(
      <MemoryRouter>
        <Lyrics match={match}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
