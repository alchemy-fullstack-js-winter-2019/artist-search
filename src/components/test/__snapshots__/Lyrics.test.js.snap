// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyrics matches a snapshot 1`] = `
Array [
  <p>
    
  </p>,
  <a
    href="/"
    onClick={[Function]}
  >
    Return to search
  </a>,
]
`;
